components/form.js

```diff
@@ -80,12 +95,28 @@ export default class Form extends React.Component {
   componentDidUpdate(prevProps, prevState) {
     localStorage.setItem("Form Fields", JSON.stringify(this.state.formFields));
     localStorage.setItem("Bearer Token", JSON.stringify(this.state.token));
+    window.addEventListener("online", this.isOnlineEvent);
   }
 
   // Handler html For when the submit button is pressed
   onSubmit = e => {
     e.preventDefault();
-    postFormData(this.state.formData, this.state.token.access_token);
+    if (navigator.onLine) {
+      postFormData(this.state.formData, this.state.token.access_token);
+    } else {
+      let formDataArray = JSON.parse(
+        localStorage.getItem("Failed Form Submission Data")
+      );
+      if (!formDataArray) {
+        formDataArray = [this.state.formData];
```

Comment:
if you define formDataArray as an array after `e.preventDefault` (around the beggining of the function), you don't need this if/else. You can just do formDataArray.push(this.state.formData)

Reply:
Sure! Thank you :)